source/slang/check.cpp

```diff
@@ -9303,6 +9337,34 @@ namespace Slang
                 attr->patchConstantFuncDecl = funcDecl;
             }
         }
+        else if (entryPoint->getStage() == Stage::Compute)
+        {
+            auto funcDecl = entryPoint->getFuncDecl();
+
+            auto params = funcDecl->GetParameters();
+
+            for (const auto& param : params)
+            {
+                if (auto semantic = param->FindModifier<HLSLSimpleSemantic>())
+                {
+                    const auto& semanticToken = semantic->name;
+
+                    String lowerName = String(semanticToken.Content).ToLower();
+
+                    if (lowerName == "sv_dispatchthreadid")
```

Comment:
This is fine as a one-off check, but it obviously exposes the fact that we aren't doing any real semantic checking on semantics (apologies for the over-use of the term "semantic" here).

Ideally I'd like us to set this up as something we can data-drive, perhaps even with semantics being something declared in the standard library, rather than needing to all be special-cased in the guts of `check.cpp`. If they were declared in the stdlib, then the mapping of a semantic to GLSL (or other targets) could hopefully follow a pattern more like what we do for builtin functions.

None of the above is an argument against the code in this PR, and more of an argument that we may want to sit down and design a systematic solution soon, if the alternative is to implement dozens of ad hoc checks like this one.